symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/jwt/JwtHelper.java

```diff
@@ -121,31 +125,58 @@ public static UserClaim validateJwt(String jwt, String certificate) throws AuthI
     }
   }
 
-	private static PrivateKey parsePKCS1PrivateKey(String pemPrivateKey) throws GeneralSecurityException {
-		try (final PemReader pemReader = new PemReader(new StringReader(pemPrivateKey))) {
-			final PemObject privateKeyObject = pemReader.readPemObject();
-			final RSAPrivateKey rsa = RSAPrivateKey.getInstance(privateKeyObject.getContent());
-			final RSAPrivateCrtKeyParameters privateKeyParameter = new RSAPrivateCrtKeyParameters(
-				rsa.getModulus(),
-				rsa.getPublicExponent(),
-				rsa.getPrivateExponent(),
-				rsa.getPrime1(),
-				rsa.getPrime2(),
-				rsa.getExponent1(),
-				rsa.getExponent2(),
-				rsa.getCoefficient()
-			);
-
-			return new JcaPEMKeyConverter().getPrivateKey(PrivateKeyInfoFactory.createPrivateKeyInfo(privateKeyParameter));
-		} catch (IOException | IllegalStateException | IllegalArgumentException e) {
-			throw new GeneralSecurityException("Unable to parse the private key. Check if the format of your private key is correct.", e);
-		}
-	}
-
-	private static PrivateKey parsePKCS8PrivateKey(String pemPrivateKey) throws GeneralSecurityException {
+  /**
+   * Extract the expiration date (in seconds) from the input jwt. If the jwt uses the Beare prefix, it
+   * will be removed before parsing. This function is not validating  the jwt signature.
+   *
+   * @param jwt to be parsed
+   * @return expiration date in seconds
+   * @throws JsonProcessingException if parsing fails
+   */
+  public static Long extractExpirationDate(String jwt) throws JsonProcessingException, AuthUnauthorizedException {
+    String claimsObj = extractDecodedClaims(dropBearer(jwt));
+    ObjectNode claims = mapper.readValue(claimsObj, ObjectNode.class);
+    return claims.get("exp").asLong();
```

Comment:
is the case of exp being absent properly failing? (or exp not being a long?)

Reply:
I will add checks in the code and in case I will throw an exception

Reply:
Done [here](https://github.com/finos/symphony-bdk-java/pull/618/commits/ae4de1bea7a5169eb62e71780b22a8946ab050c0)